### ISSUE TITLE
Rename Source Fields When Creating Stream/Table

### ISSUE THREAD

**Issue by jnadler** [confluentinc/ksql#785]
Source formats like JSON are tolerant of pretty much any name for a field, even including special characters that would not be valid in a SQL field name.

Allow users to provide a mapping from the source field names (in the JSON or Avro source) to SQL-compliant fieldnames.   

In slack #ksql, @hjafarpour suggests "I think we can include this in the `WITH` clause for `CREATE STREAM` and `CREATE TABLE` ddl statements and provide a mapping from the existing names to SQL compliant names."

**Comment by rmoff** [comment]
I think #677 will negate the need for this, what do you think @jnadler ?

**Comment by jnadler** [comment]
Sounds like it.

**Comment by rmoff** [comment]
👍 closing.

**Comment by analytik** [comment]
I still think renaming fields is a worthwhile endeavor. For example we use Debezium to stream DB changes into Kafka, including UPDATEs, so the original and new DB row itself is wrapped in a `before` and `after` JSON field, e.g.

```
ksql> PRINT 'foo_db.public.users' FROM BEGINNING;
Format:JSON
{"ROWTIME":1537963287332,
"ROWKEY":"{\"id\":\"123qwer\"}",
"before":{"id":"123qwer","created":"2018-05-24T14:34:07.335Z","updated":"2018-05-24T14:34:07.335Z","balance":5000.0, "currency":"EUR"},
"after":{"id":"123qwer","created":"2018-05-24T14:34:07.335Z","updated":"2018-05-24T22:22:22.335Z","balance":10000.0, "currency":"EUR"},
"source":{"version":"0.9.0.Alpha1", "name":"foo_db", "ts_usec":1537963286085000, "txId":5777, "lsn":6781059072, "snapshot":true, "last_snapshot_record":false}, "op":"u", "ts_ms":1537963286086}
```

I don't always want to have to access a struct, in most times I only care about `after` values, and in other cases I'd just want to flatten the structure and rename fields to something more relevant and readable, e.g. `balance_before` and `balance_after`.
